repository/src/main/java/org/alfresco/repo/rule/RuleServiceImpl.java

```diff
@@ -459,7 +459,13 @@ public boolean isRuleTypeEnabled(String ruleType)
     public boolean hasRules(NodeRef nodeRef)
     {
         return getRules(nodeRef).size() != 0;
-    } 
+    }
+
+    @Override
+    public boolean hasNonInheritedRules(NodeRef nodeRef)

```

Comment:
Silly question: How does adding this method fix the issue?

Reply:
Inside the method that does the actual linking `linkToRuleSet`, we are checking whether the folder that is trying to link has any pre-existing rules. With this method, we will ignore inherited rules in that check and perform the linking